### PR TITLE
MIGENG-31: UX Implementation - Workload Inventory Report

### PR DESCRIPTION
**Pull request by carlosthe19916**
This PR added the "Workload inventory report" as you can see in the screenshot below:

The "Export as CSV" button is not working. Waiting for implementation in the backend.
This PR does not include the filtering feature because probably it will not be part of the current sprint.

61797043-f261d980-ae26-11e9-97f6-10ca4a1750da

### CODE REVIEW

**Comment by carlosthe19916** [comment]
My last commit added functionality to the "Download as CSV" button. Not it will download the file using .csv extension.

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx [hunk 1]
```diff
@@ -1,27 +1,487 @@
...
+                },
+                {
+                    title: 'Cluster',
+                    key: 'cluster',
+                    props: {
+                        className: 'vertical-align-middle'
+                    }
+                },
+                {
+                    title: 'VMware',
```
Comment (mrizzi) [inline src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]:
```suggestion
                    title: 'VM name',
```

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx [hunk 2]
```diff
@@ -1,27 +1,487 @@
...
+                {
+                    title: 'OS type',
+                    key: 'osName',
+                    props: {
+                        className: 'vertical-align-middle'
+                    },
+                    transforms: [ sortable ]
+                },
+                {
+                    title: 'Efford',
```
Comment (mrizzi) [inline src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]:
```suggestion
                    title: 'Effort',
```

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx [hunk 3]
```diff
@@ -1,27 +1,487 @@
...
+                    key: 'complexity',
+                    props: {
+                        className: 'vertical-align-middle'
+                    },
+                    transforms: [ sortable ]
+                },
+                {
+                    title: 'Recomended targets',
```
Comment (mrizzi) [inline src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]:
```suggestion
                    title: 'Recommended targets',
```
Here, in the mock ups, `Recommended` and `targets` are in 2 lines, not in the same one, can we do it?

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx [hunk 4]
```diff
@@ -1,27 +1,487 @@
...
+                    key: 'recommendedTargetsIMS',
+                    props: {
+                        className: 'vertical-align-middle'
+                    }
+                },
+                {
+                    title: 'Flag IMS',
```
Comment (mrizzi) [inline src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]:
```suggestion
                    title: 'Flags IMS',
```

**Comment by carlosthe19916** [comment]
> There are some typos in the table headings.
> Other than this, when i select the ICS report, for 1-2 seconds, the following content appears
> Screenshot from 2019-08-01 17-02-38
> and then it disappears and i can see the usual list of analysis.
> WDYT?

@mrizzi Thanks for reporting those problems. My last commits fixed them.

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx (hunk 1 above)
Comment (carlosthe19916) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: Requested changes applied

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx (hunk 2 above)
Comment (carlosthe19916) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: Requested changes applied

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx (hunk 4 above)
Comment (carlosthe19916) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: Requested changes applied

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx (hunk 3 above)
Comment (carlosthe19916) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: I've tried using Patternfly styles but I couldn't do it. I really like the way it is currently implemented (just one line); let me ask this to the UX team

Comment (mrizzi) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: Ok, great, thanks.
Let's wait for their opinion to understand if we can keep it as it is or we need to change it

**Comment by mrizzi** [comment]
> @mrizzi Thanks for reporting those problems. My last commits fixed them.
I've tested it and it works great now :+1: 
Let's wait to get an answer for the `Recommended targets` heading.

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx (hunk 3 above)
Comment (doruskova) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: I don't have a problem with a one line. It seems to be ok.

**Comment by carlosthe19916** [comment]
@doruskova Thanks for discovering the problem: "table is not sorting by EFFORT column". My last commit fixed that error:

Before (effort column has no sorting)
Screenshot from 2019-08-02 11-37-47

After
Screenshot from 2019-08-02 12-44-24

File: src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx (hunk 3 above)
Comment (mrizzi) [reply src/pages/ReportView/WorkloadInventory/WorkloadInventory.tsx]: Thanks @doruskova, let's keep it as it is, no further change required :+1:

**Comment by carlosthe19916** [comment]
> @carlosthe19916 i was reviewing [project-xavier/xavier-integration#27](https://github.com/project-xavier/xavier-integration/pull/27) and the UI is not working as expected.
> I sorted by `VM name` field and the first page of results looked good.
> The remote call don has been `/api/xavier/report/338/workload-inventory?page=0&size=10&orderBy=vmName&orderAsc=true`
> Then i moved to page 2 and the sorting was wrong because the remote call had another `orderBy` field `/api/xavier/report/338/workload-inventory?page=1&size=10&orderBy=workload&orderAsc=true`.
> 
> Could you have a look to this issue please?
> Thanks

@mrizzi Thanks for discovering that problem. My last commit solved the problem.
The cause of the problem is that collapse tables always add an extra column so the index of every should be column is `index-1` and not `index`

**Comment by carlosthe19916** [comment]
> Thanks @carlosthe19916: the commit solved the issue.
> I found another thing which is not a bug but a requirement we had: first time you sort by a column the page should be reset to be the first one (just like we said will happen with filtering).
> Right now, if i'm in page 3 and i sort by a column, i will still see page 3 instead of page 1.

@mrizzi I've just added a new commit which will reset the page to 1 every time the user sorts the table.

I also needed to upgrade the npm @patternfly/react-core version to 3.82.2 because of this issue https://github.com/patternfly/patternfly-react/issues/2629

**Pull request merged**